test-packages/framework/src/framework.ts

```diff
@@ -134,23 +134,32 @@ export class TestFramework implements TestFrameworkAPI {
   ): Promise<ReadFileResult> {
     const root = this.getProjectRoot();
     const filePath = join(root, ...pathSegments);
-    if (!fileExits(filePath)) {
+    if (!(await fileExits(filePath))) {
```

Comment:
Instead of checking if file exists it is better to check file read failed, because these operations are async it could be that a file exists for the first call, but doesn't for the second and you end up with an unhandled error anyways. https://nodejs.org/api/fs.html#fsexistspath-callback